### PR TITLE
Fix GloboIE.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/991

### CODE REVIEW

**Comment by renatobusatto** [comment]
Your solution is working perfectly, thank you very much!

File: yt_dlp/extractor/globo.py [hunk 1]
```diff
@@ -26,18 +25,18 @@ class GloboIE(InfoExtractor):
     _NETRC_MACHINE = 'globo'
     _TESTS = [{
         'url': 'http://g1.globo.com/carros/autoesporte/videos/t/exclusivos-do-g1/v/mercedes-benz-gla-passa-por-teste-de-colisao-na-europa/3607726/',
-        'md5': 'b3ccc801f75cd04a914d51dadb83a78d',
+        'md5': 'a0e8cbde65ce5f74ff87616490b10262',
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:28]: md5s that keep changing are useless to test for. `skip_download` the tests

File: yt_dlp/extractor/globo.py [hunk 2]
```diff
@@ -102,73 +77,66 @@ def _real_extract(self, url):
...
+        md5_data = (received_md5 + padded_sign_time + '0xAC10FD').encode()
+        signed_md5 = base64.urlsafe_b64encode(hashlib.md5(md5_data).digest()).decode().strip('=')
+        signed_hash = hash_prefix + padded_sign_time + signed_md5
+        source = security['source']['url_parts']
+        resource_url = source['scheme'] + '://' + source['domain'] + source['path']
+        signed_url = '%s?h=%s&k=html5&a=%s' % (resource_url, signed_hash, 'F' if video.get('subscriber_only') else 'A')
+
+        formats.extend(self._extract_m3u8_formats(
+                       signed_url, video_id, 'mp4', entry_protocol='m3u8_native',
+                       m3u8_id='hls', fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:120]:
```suggestion
        formats.extend(self._extract_m3u8_formats(
            signed_url, video_id, 'mp4', entry_protocol='m3u8_native', m3u8_id='hls', fatal=False))
```

The intends are confusing. The params are inside `_extract_m3u8_formats` but are aligned to the `formats.extend`. weird that flake8 doesnt catch this

File: yt_dlp/extractor/globo.py [hunk 3]
```diff
@@ -102,73 +77,66 @@ def _real_extract(self, url):
...
                 })
-                continue
-
-            security = self._download_json(
-                'http://security.video.globo.com/videos/%s/hash' % video_id,
-                video_id, 'Downloading security hash for %s' % resource_id, query={
-                    'player': 'desktop',
-                    'version': '5.19.1',
-                    'resource_id': resource_id,
+        for sub_lang, sub_url in try_get(security, lambda x: x['source']['subtitles'], expected_type=dict).items():
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:129]: the try_get may return None

File: yt_dlp/extractor/globo.py [hunk 4]
```diff
@@ -102,73 +77,66 @@ def _real_extract(self, url):
...
-                    signed_url, resource_id, mpd_id='dash', fatal=False))
-            elif resource_id.endswith('manifest') or resource_url.endswith('/manifest'):
-                formats.extend(self._extract_ism_formats(
-                    signed_url, resource_id, ism_id='mss', fatal=False))
-            else:
-                formats.append({
-                    'url': signed_url,
-                    'format_id': 'http-%s' % resource_id,
-                    'height': int_or_none(resource.get('height')),
+        for sub_lang, sub_url in try_get(security, lambda x: x['source']['subtitles_webvtt'], expected_type=dict).items():
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:134]: same

**Comment by pukkandan** [comment]
feel free to merge once these are addressed @Ashish0804

File: yt_dlp/extractor/globo.py (hunk 3 above)
Comment (Ashish0804) [reply yt_dlp/extractor/globo.py:129]: It shouldn't really matter here as missing `source` will be caught before and `subtitles` and `subtitles_webvtt` are always present, even as an empty dict.

But still do you recommend i do something like
```py
subs = try_get
if subs:
  for sub_lang, sub_url in subs.items():
    append
```

Comment (pukkandan) [reply yt_dlp/extractor/globo.py:129]:
```suggestion
        subs = try_get(security, lambda x: x['source']['subtitles'], expected_type=dict) or {}
        for sub_lang, sub_url in subs.items():
```

**Pull request merged**